event-protocol/schemas/gherkin-document.json

```diff
@@ -0,0 +1,59 @@
+{
+  "id": "https://raw.github.com/cucumber/cucumber/master/event-protocol/schemas/gherkin-document.json#",
+  "title": "gherkin-document",
+  "description": "Describes the Abstract Syntax Tree of a Gherkin Document",
+  "type": "object",
+  "properties": {
+    "type": {
+      "type": "string"
+    },
+    "uri": {
+      "type": "string"
+    },
+    "document": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string"
+        },
+        "feature": {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string"
+            },
+            "tags": {
+              "type": "array",
+              "items": { "$ref": "defs.json#/ast-tag" }
+            },
+            "location": {
+              "type": "object",
+              "items": { "$ref": "defs.json#/ast-location" }
+            },
+            "language": {
+              "type": "string"
+            },
+            "keyword": {
+              "type": "string"
+            },
+            "name": {
+              "type": "string"
+            },
+            "children": {
+              "type": "array"
```

Comment:
Ideally we should, yes. As mentioned above, the elements can be one of 3 types, so I think we need to use something like [oneOf](http://json-schema.org/latest/json-schema-validation.html#rfc.section.5.24). I was lazy and decided that super strict validation of `gherkin-document` is less important than all of the other schemas, because `gherkin-document` objects will practically only be generated by the Gherkin library, and that's *really* thoroughly tested.

That said, I think it would be nice to have a full schema - a schema is useful for more than validation - it can also be used to generate documentation. See http://json-schema.org/implementations.html under "Documentation generation". Especially [Doca](https://github.com/cloudflare/doca/) looks interesting.